cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -474,9 +479,17 @@ func (s *Server) Index(args *indexArgs) (state indexState, err error) {
 	if args.Incremental {
 		bo := args.BuildOptions()
 		bo.SetDefaults()
-		if bo.IncrementalSkipIndexing() {
+		state := bo.IndexState()
```

Comment:
Should we log "IndexStateCorrupt", "IndexStateMissing"?

Reply:
nit: it is a bit confusing that we have 2 types of indexState, one from `build` and one from indexserver and the `build.IndexState` shadows the named return parameter `state`.

Reply:
I agree. The worst part is the shadowing, I will fix that. But build.IndexState is different to this indexState, so I think the similiar name is fine given namespacing. Do you have  a proposal for a better name?

IndexStateMissing is to be expected. Will always happen for a new repo. IndexStateCorrupt isn't. I will add a log line for corrupt since it is unexpected.